File: employeesvacationtracker.client/src/models/entities/employee.ts
export interface IEmployee {
  id: number; // identifier
  jobTitle: string;
  departement: keyof typeof Departments;
  role: string;
  joiningDate: string;
  userId: string;
}

export enum Departments {
  Development,
  Design,
  Testing,
  HR,
}

export interface CreateOrUpdateEmployeeFormData {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  jobTitle: string;
  department: Departments;
  joiningDate: Date;
}

export interface CreateOrUpdateEmployeeRequest {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  jobTitle: string;
  department: Departments;
  joiningDate: string;
}

File: EmployeesVacationTracker.DomainLayer/Entities/Employee.cs
using EmployeesVacationTracker.DomainLayer.Common;

namespace EmployeesVacationTracker.DomainLayer.Entities
{
    public enum EmployeeDepartment
    {
        Development,
        Design,
        Testing,
        HR
    }

    public class Employee : BaseEntity
    {
        public string JobTitle { get; set; }
        public EmployeeDepartment Department { get; set;}
        public DateTimeOffset JoiningDate { get; set; }
        public int UserId { get; set; }
    }
}
